hudi-spark/src/main/scala/org/apache/hudi/IncrementalRelation.scala

```diff
@@ -100,17 +101,22 @@ class IncrementalRelation(val sqlContext: SQLContext,
         .get, classOf[HoodieCommitMetadata])
       fileIdToFullPath ++= metadata.getFileIdAndFullPaths(basePath).toMap
     }
+    val pathGlobPattern = optParams.getOrElse(DataSourceReadOptions.INCR_PATH_GLOB_OPT_KEY, "")
+    val filteredFullPath = if(!pathGlobPattern.equals("")) {
+      val globMatcher = new GlobPattern("*" + pathGlobPattern)
```

Comment:
should we leave the `*` to the user? i.e let the user pass in `*` if needed? or is that needed for the matching...

I am not familiar with this class per se.. 

Also http://hadoop.apache.org/docs/r2.8.0/api/allclasses-noframe.html does not seem to have `GlobPattern` is this class still around.. Was a bit confused by that..

Reply:
the path here is a full HDFS path so we need `*` here to match with the prefix.  The benefit if we include `*` here is that the user will have a consistent interface. When loading the full table, they will do `.load(basePath + "/2016/*/*/*")` and in incremental pulling the `String` the user defined will be the same. If we leave the `*` to the user I think it might cause some confusion there and the users need to read this part of the code themselves to fully understand how things work here.  
Yea I couldn't find any documents as well. The `GlobFilter` in the API list is using `GlobPattern` inside https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/GlobFilter.java#L67 and the class is still around https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/GlobPattern.java